### PR TITLE
Implement OpenID Connect testing

### LINKED ISSUE #51: OpenID Connect Tests
Currently there are no OpenID Connect tests. We will need to decide where these should be placed.  Perhaps they are best suited within existing launch test sequences.

### LINKED ISSUE #103: RestClient.get 404 error
`RestClient.get` in `LoggedRestClient.rb` (https://github.com/fhir-crucible/crucible_smart_app/blob/master/lib/utils/LoggedRestClient.rb#L36) throws an error when it encounters a 404, rather than causing the test to fail. A failure should display in the application in this case, rather than an error.

### PR DESCRIPTION
**Pull request by kjmahalingam**
This PR implements OpenID Connect tests (resolves #51).

resolves #103

### CODE REVIEW

**Comment by kjmahalingam** [comment]
Consulted @jricher about OpenID testing and made changes based on suggestions and feedback.

Also fixed #103.

File: lib/sequences/03_patient_standalone_launch_sequence.rb [hunk 1]
```diff
@@ -111,6 +111,11 @@ class PatientStandaloneLaunchSequence < SequenceBase
     @instance.save!
     @instance.update(scopes: scopes)
 
+    if @token_response_body.has_key?('id_token')
```
Comment (arscan) [inline lib/sequences/03_patient_standalone_launch_sequence.rb]: Should we clear the id_token if it doesn't exist (for ehr launch too)

File: lib/sequences/05_openid_connect_sequence.rb [hunk 2]
```diff
@@ -0,0 +1,111 @@
+class OpenIDConnectSequence < SequenceBase
```
Comment (arscan) [inline lib/sequences/05_openid_connect_sequence.rb]: Add sequence self-test.  We need to have all these done this week, so might as well get that in here now.

File: lib/sequences/03_patient_standalone_launch_sequence.rb (hunk 1 above)
Comment (arscan) [reply lib/sequences/03_patient_standalone_launch_sequence.rb]: Noticed we kinda do this on the modal post, so maybe that is sufficient.

**Comment by kjmahalingam** [comment]
Changes requested by @arscan have been addressed.

**Pull request merged**